cpu/kinetis/periph/timer.c

```diff
@@ -628,7 +628,7 @@ int timer_init(tim_t dev, unsigned long freq, timer_cb_t cb, void *arg)
 
 int timer_set(tim_t dev, int channel, unsigned int timeout)
 {
-    if (channel != 0) {
+    if (channel < 0 || channel >= (int)TIMER_CHANNEL_NUMOF) {
```

Comment:
```suggestion
    if ((unsigned)channel >= TIMER_CHANNEL_NUMOF) {
```

When casting a signed integer in two's complement representation to an unsigned integers, negative values become huge numbers. With that in mind, this can be simplified. (I'm aware that the C standard does allow for representations other than two's complement with the exception of atomic types, but no one has ever seen hardware not using two's complement.)